File: 11. Container With Most Water/Sol_1.ts
var maxArea = (height: number[]): number => {
    let right: number = height.length - 1
    let left: number = 0
    let area: number = 0

    while (left < right) {
        let curr_area: number = Math.min(height[left], height[right]) * (right - left)
        area = Math.max(area, curr_area)

        if (height[left] < height[right]) left += 1
        else right -= 1
    }
    return area
}

File: 11. Container With Most Water/Sol_1.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {

        int left = 0;
        int right = height.size() - 1;
        int area = 0;

        while (left < right)
        {
            int curr_area = min(height[left], height[right]) * (right - left);
            area = max(area, curr_area);

            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return area;
    }
};